DataCapturingTests/DataCapturingTests.swift

```diff
@@ -329,4 +329,26 @@ class DataCapturingTests: XCTestCase {
         print(locationsCount)
         XCTAssertTrue(locationsCount>=5)
     }
+
+    func testStartPauseResumeStop_DistanceCalculationContainsLastSegment() throws {
```

Comment:
This test was a reproducer for a possible bug reported by Constantin. I added it here even though the bug was not reproducible. Actually the test should not really be part of this branch. I can remove it if you like.